# Analysis

### 1. Import thư viện cần thiết và load data

In [7]:
import pandas as pd
from sklearn.metrics import (
    mean_absolute_error,
    root_mean_squared_error,
    r2_score
)

true_target = pd.read_csv("../data/preprocessed/data_test.csv")["throughput_down_mbps"]
pred_baseline = pd.read_csv("../data/prediction-baseline/prediction_baseline.csv")
pred_linear_regression = pd.read_csv("../data/prediction-linear-regression/prediction_linear_regression.csv")

### 2. Tạo bảng so sánh giữa hai mô hình Mean và Linear Regression

In [8]:
models_performance = pd.DataFrame({
    'Model': ['Baseline (Mean)', 'Linear Regression'],
    'MAE': [mean_absolute_error(true_target, pred_baseline), mean_absolute_error(true_target, pred_linear_regression)],
    'RMSE': [root_mean_squared_error(true_target, pred_baseline), root_mean_squared_error(true_target, pred_linear_regression)],
    'R^2': [r2_score(true_target, pred_baseline), r2_score(true_target, pred_linear_regression)]
})

display(models_performance)

,Model,MAE,RMSE,R^2
0,Baseline (Mean),40.676235,42.340932,-0.003444
1,Linear Regression,7.203437,10.900410,0.933494


#### Ý nghĩa
- Mean Absolute Error (MAE): Trên trung bình, dự đoán của mô hình Linear Regression lệch khỏi giá trị đúng ~7.2 Mbps, rất gần so với mức lệch ~40.68 Mbps của mô hình Mean.
- Root Mean Squared Error (RMSE): Mô hình Linear Regression lệch khỏi giá trị đúng ít hơn mô hình Mean.
- Coefficient of determination (R^2): Mô hình Linear Regression phản ánh tốt sự biến thiên của thông lượng, trong khi mô hình Mean hầu như không phản ánh được điều này.